lib/matplotlib/backend_tools.py

```diff
@@ -99,20 +98,6 @@ def trigger(self, sender, event, data=None):
 
         pass
 
-    @figure.setter
-    def figure(self, figure):
-        """
-        Set the figure
-
-        Set the figure to be affected by this tool
-
-        Parameters
-        ----------
-        figure: `Figure`
-        """
-
-        self._figure = figure
```

Comment:
To make it clear I don't mind making the change from ``ToolManager(self.figure.canvas)`` to ``ToolManager(self.figure)`` here.  I think that makes more sense and exists in the overlap of MEP27 and MEP22.  Then again, perhaps easier to do it all in one go in a future PR, especially as this won't affect the backends... but your call on that.